components/icon-picker/inline-icon-picker.tsx

```diff
@@ -1,46 +1,40 @@
-import PropTypes from 'prop-types';
 import styled from '@emotion/styled';
 import { Dropdown } from '@wordpress/components';
 import { useCallback } from '@wordpress/element';
 
+import { DropdownProps } from '@wordpress/components/build-types/dropdown/types';
+
+// extract type from DropdownProps renderToggle function props
+type RenderToggleProps = Parameters<DropdownProps['renderToggle']>[0];
+
 import { IconPicker } from './icon-picker';
 import { Icon } from './icon';
+import React from 'react';
+import type { FC } from 'react';
+import type { IconPickerProps } from './icon-picker';
 
 const StyledIconPickerDropdown = styled(IconPicker)`
 	margin: 6px;
 	width: 306px;
 `;
 
-/**
- * InlineIconPicker
- *
- * @typedef InlineIconPickerProps
- * @property {string} buttonLabel label
- *
- * @param {InlineIconPickerProps} props InlineIconPickerProps
- * @returns {*}
- */
-export const InlineIconPicker = (props) => {
+export const InlineIconPicker: FC<IconPickerProps> = (props) => {
 	const { value, ...rest } = props;
 	const IconButton = useCallback(
-		({ onToggle }) => (
-			<Icon name={value?.name} iconSet={value?.iconSet} onClick={onToggle} {...rest} />
+		({ onToggle }: RenderToggleProps) => (
+			<Icon name={value.name} iconSet={value?.iconSet} onClick={onToggle} {...rest} />
```

Comment:
@fabiankaegy the `iconSet` property has not been defined as optional in the [`Icon` type definition](https://github.com/10up/block-components/pull/191/files#diff-944cf70fc1be5c7b4fc19291f6b09dba58a5fed031b77f7bde5061a80a8b6afeR4). You can probably remove the question mark here as you did with the `name` property.